rules/library.bzl

```diff
@@ -219,6 +220,82 @@ def _canonicalize_swift_version(swift_version):
 
     return swift_version
 
+def _xcframework_build_type(*, linkage, packaging):
+    return (linkage, packaging)
+
+def _xcframework_slice(*, xcframework_name, identifier, platform, platform_variant, supported_archs, build_type, path):
+    linkage, packaging = _xcframework_build_type(**build_type)
+    import_name = "{}-{}".format(xcframework_name, identifier)
+    if (linkage, packaging) == ("dynamic", "framework"):
+        apple_dynamic_framework_import(
+            name = import_name,
+            framework_imports = native.glob([path + "/**/*"]),
+            deps = [],
+            tags = _MANUAL,
+        )
+    elif (linkage, packaging) == ("static", "framework"):
+        apple_static_framework_import(
+            name = import_name,
+            framework_imports = native.glob([path + "/**/*"]),
+            deps = [],
+            tags = _MANUAL,
+        )
+    elif (linkage, packaging) == ("static", "library"):
+        objc_import(
+            name = import_name,
+            archives = [path],
+            tags = _MANUAL,
+        )
+    else:
+        fail("Unsupported xcframework slice type {} ({}) in {}".format(
+            build_type,
+            identifier,
+            xcframework_name,
+        ))
+    return (platform, platform_variant, supported_archs, import_name)
+
+def _xcframework(*, library_name, name, slices):
+    xcframework_name = "{}-import-{}.xcframework".format(library_name, name)
+    conditions = {}
+    for slice in slices:
+        platform, platform_variant, archs, name = _xcframework_slice(xcframework_name = xcframework_name, **slice)
+        platform_setting = "@build_bazel_rules_ios//rules/apple_platform:" + platform
+        # platform_variant_setting = "@build_bazel_rules_ios//rules/apple_platform:platform_variant_" + platform_variant if platform_variant else None
+
+        for arch in archs:
+            if arch == "arm64" and platform_variant == "simulator":
+                # TODO: support sim on apple silicon by having a config setting for platform_variant
+                continue
+
+            arch_setting = "@build_bazel_rules_apple//apple:{}_{}".format(platform, arch)
```

Comment:
Any reason to not put this in a select?

Reply:
it's used in the condition of the select, created by the `selects` library in skylib